<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [Twisted-Python] how to write a safe catch-all
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:twisted-python%40twistedmatrix.com?Subject=%5BTwisted-Python%5D%20how%20to%20write%20a%20safe%20catch-all&In-Reply-To=4CA36675.5010603%40imperial.ac.uk">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="022936.html">
   <LINK REL="Next"  HREF="022938.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Twisted-Python] how to write a safe catch-all</H1>
    <B>Chris Withers</B> 
    <A HREF="mailto:twisted-python%40twistedmatrix.com?Subject=%5BTwisted-Python%5D%20how%20to%20write%20a%20safe%20catch-all&In-Reply-To=4CA36675.5010603%40imperial.ac.uk"
       TITLE="[Twisted-Python] how to write a safe catch-all">chris at simplistix.co.uk
       </A><BR>
    <I>Wed Sep 29 13:16:30 EDT 2010</I>
    <P><UL>
        <LI>Previous message: <A HREF="022936.html">[Twisted-Python] how to write a safe catch-all
</A></li>
        <LI>Next message: <A HREF="022938.html">[Twisted-Python] how to write a safe catch-all
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#22937">[ date ]</a>
              <a href="thread.html#22937">[ thread ]</a>
              <a href="subject.html#22937">[ subject ]</a>
              <a href="author.html#22937">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>On 29/09/2010 17:16, Phil Mayers wrote:
&gt;<i> No.
</I>&gt;<i>
</I>&gt;<i> The problem is that your example is malformed.
</I>
Well, it's not, it's the reality of the situation and one I'd like to 
protect against; &quot;the scheduler must not die&quot; is the rule I need to make 
work...

&gt;<i> You do this:
</I>&gt;<i>
</I>&gt;<i>    1. Create a deferred on the &quot;Break&quot; class instance
</I>
The &quot;Break&quot; class, in reality, is the bowels of 
twisted.protocols.ftp.FTPClient, copying from my previous mail:

      self.failIfNotConnected(error.getConnectError((err, strerror(err))))
    File &quot;Twisted-8.2.0-py2.5-linux-x86_64.egg/twisted/internet/tcp.py&quot;,
line 588, in failIfNotConnected
      del self.connector
--- &lt;exception caught here&gt; ---
    File &quot;ourcode.py&quot;, line 180, in checkSchedule
      yield self.sendTransmissions(...)
exceptions.GeneratorExit:

How can I protect my scheduler against code which doesn't catch an 
exception when it should?

cheers,

Chris

</PRE>




<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="022936.html">[Twisted-Python] how to write a safe catch-all
</A></li>
	<LI>Next message: <A HREF="022938.html">[Twisted-Python] how to write a safe catch-all
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#22937">[ date ]</a>
              <a href="thread.html#22937">[ thread ]</a>
              <a href="subject.html#22937">[ subject ]</a>
              <a href="author.html#22937">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://twistedmatrix.com/cgi-bin/mailman/listinfo/twisted-python">More information about the Twisted-Python
mailing list</a><br>
</body></html>
